examples/convert_osi2mcap.cpp

```diff
@@ -0,0 +1,205 @@
+    {"SensorView", osi3::ReaderTopLevelMessage::kSensorView},          {"HostVehicleData", osi3::ReaderTopLevelMessage::kHostVehicleData},
+    {"TrafficCommand", osi3::ReaderTopLevelMessage::kTrafficCommand},  {"TrafficCommandUpdate", osi3::ReaderTopLevelMessage::kTrafficCommandUpdate},
+    {"TrafficUpdate", osi3::ReaderTopLevelMessage::kTrafficUpdate},    {"MotionRequest", osi3::ReaderTopLevelMessage::kMotionRequest},
+    {"StreamingUpdate", osi3::ReaderTopLevelMessage::kStreamingUpdate}};
+
+void printHelp() {
+    std::cout << "Usage: convert_osi2mcap <input_file> <output_file> [--input-type <message_type>]\n\n"
+              << "Arguments:\n"
+              << "  input_file              Path to the input OSI trace file\n"
+              << "  output_file             Path to the output MCAP file\n"
+              << "  --input-type <message_type>   Optional: Specify input message type if not stated in filename\n\n"
+              << "Valid message types:\n";
+    for (const auto& [type, _] : kValidTypes) {
+        std::cout << "  " << type << "\n";
+    }
+}
+
+std::optional<ProgramOptions> parseArgs(const int argc, const char** argv) {
+    if (argc < 3 || std::string(argv[1]) == "--help" || std::string(argv[1]) == "-h") {
+        printHelp();
+        return std::nullopt;
+    }
+
+    ProgramOptions options;
+    options.input_file_path = argv[1];
+    options.output_file_path = argv[2];
+    options.message_type = osi3::ReaderTopLevelMessage::kUnknown;
+
+    for (int i = 3; i < argc; i++) {
+        if (std::string arg = argv[i]; arg == "--input-type" && i + 1 < argc) {
+            const std::string type_str = argv[++i];
+            auto types_it = kValidTypes.find(type_str);
+            if (types_it == kValidTypes.end()) {
+                std::cerr << "Error: Invalid message type '" << type_str << "'\n\n";
+                printHelp();
+                return std::nullopt;
+            }
+            options.message_type = types_it->second;
+        }
+    }
+
+    return options;
+}
+
+int main(const int argc, const char** argv) {
+    const auto options = parseArgs(argc, argv);
+    if (!options) {
+        return 1;
+    }
+
+    std::cout << "Input file: " << options->input_file_path << std::endl;
+    std::cout << "Output file: " << options->output_file_path << std::endl;
+
+    auto trace_file_reader = osi3::SingleChannelBinaryTraceFileReader();
+    if (!trace_file_reader.Open(options->input_file_path, options->message_type)) {
+        std::cerr << "ERROR: Could not open input file " << options->input_file_path << std::endl;
+        return 1;
+    }
+
+    auto trace_file_writer = osi3::MCAPTraceFileWriter();
+    if (!trace_file_writer.Open(options->output_file_path)) {
+        std::cerr << "ERROR: Could not open output file " << options->output_file_path << std::endl;
+        return 1;
+    }
+
+    // according to the OSI specification mcap must contain a timestamp and zero_time metadata entry
```

Comment:
This is not true anymore. zero_time is no longer mandatory. Trying to parse it from the filename is still a good idea.